Signal/src/ViewControllers/NewContactThreadViewController.m

```diff
@@ -295,46 +320,166 @@ - (void)updateTableContents
-                                        viewController.nonContactConversationDelegate = weakSelf;
-                                        [weakSelf.navigationController pushViewController:viewController animated:YES];
-                                    }]];
+    [staticSection
+        addItem:[OWSTableItem disclosureItemWithText:NSLocalizedString(@"NEW_CONVERSATION_FIND_BY_PHONE_NUMBER",
+                                                         @"A label the cell that lets you add a new member to a group.")
+                                     customRowHeight:self.actionCellHeight
+                                         actionBlock:^{
+                                             NewNonContactConversationViewController *viewController =
+                                                 [NewNonContactConversationViewController new];
+                                             viewController.nonContactConversationDelegate = weakSelf;
+                                             [weakSelf.navigationController pushViewController:viewController
+                                                                                      animated:YES];
+                                         }]];
 
     if (self.contactsViewHelper.contactsManager.isSystemContactsAuthorized) {
         // Invite Contacts
-        [section
+        [staticSection
             addItem:[OWSTableItem
                         disclosureItemWithText:NSLocalizedString(@"INVITE_FRIENDS_CONTACT_TABLE_BUTTON",
                                                    @"Label for the cell that presents the 'invite contacts' workflow.")
-                               customRowHeight:kActionCellHeight
+                               customRowHeight:self.actionCellHeight
                                    actionBlock:^{
                                        [weakSelf presentInviteFlow];
                                    }]];
     }
+    [contents addSection:staticSection];
+
+    BOOL hasSearchText = [self.searchBar text].length > 0;
+
+    if (hasSearchText) {
+        for (OWSTableSection *section in [self contactsSectionsForSearch]) {
+            [contents addSection:section];
+        }
+    } else {
+        for (OWSTableSection *section in [self collatedContactsSections]) {
+            [contents addSection:section];
+        }
+        contents.sectionForSectionIndexTitleBlock = ^NSInteger(NSString *_Nonnull title, NSInteger index) {
+            // We have at least one non-collated section - the "Find By Phone Number / Inbite Friends" section
+            // If we don't account for these, our collation indices will be offset.
+
+            // Count the sections which are not part of our collation.
+            NSInteger extraSections
+                = (NSInteger)contents.sections.count - (NSInteger)self.collation.sectionTitles.count;
+
+            NSInteger sectionIndex = [self.collation sectionForSectionIndexTitleAtIndex:index] + extraSections;
```

Comment:
I guess I also wonder if we add a `NSNumber *collatedSectionIndex` property to `OWSTableSection`, would that simplify things?

Reply:
I'm not totally sure what you mean - would that entail adding a map of `(collatedSectionIndex) -> (OWSTableSection)` in OWSTableView?

I'm going to leave it as is for now, but happy to revisit.